php-checks/src/main/java/org/sonar/php/checks/MultilineBlocksCurlyBracesCheck.java

```diff
@@ -26,104 +26,162 @@
     List<StatementTree> statements = getStatements(tree);
 
     for (int i = 0; i < statements.size() - 1; i++) {
       StatementTree currentStatement = statements.get(i);
-
       if (currentStatement.is(Kind.IF_STATEMENT)) {
-        checkStatement(getLastStatement((IfStatementTree)currentStatement), i, statements);
-
+        IfStatementTree ifStatement = (IfStatementTree) currentStatement;
+        checkStatement(ifStatement.ifToken(), ifStatement.condition(), getLastStatementOfIf(ifStatement), i, statements);
       } else if (currentStatement.is(Kind.FOR_STATEMENT)) {
-        checkStatement(((ForStatementTree) currentStatement).statements().get(0), i, statements);
-
+        ForStatementTree forStatement = (ForStatementTree) currentStatement;
+        checkStatement(forStatement.forToken(), forStatement.closeParenthesisToken(), forStatement.statements().get(0), i, statements);
       } else if (currentStatement.is(Kind.FOREACH_STATEMENT)) {
-        checkStatement(((ForEachStatementTree) currentStatement).statements().get(0), i, statements);
-
+        ForEachStatementTree forStatement = (ForEachStatementTree) currentStatement;
+        checkStatement(forStatement.foreachToken(), forStatement.closeParenthesisToken(), forStatement.statements().get(0), i, statements);
       } else if (currentStatement.is(Kind.WHILE_STATEMENT)) {
-        checkStatement(((WhileStatementTree) currentStatement).statements().get(0), i, statements);
-
+        WhileStatementTree whileStatement = (WhileStatementTree) currentStatement;
+        checkStatement(whileStatement.whileToken(), whileStatement.condition(), whileStatement.statements().get(0), i, statements);
       }
+    }
+  }
 
+  private static StatementTree getLastStatementOfIf(StatementTree statement) {
+    if (!statement.is(Kind.IF_STATEMENT)) {
+      return statement;
     }
+    List<StatementTree> childStatements;
+    IfStatementTree ifStatement = (IfStatementTree) statement;
+    ElseClauseTree elseClause = ifStatement.elseClause();
+    List<ElseifClauseTree> elseifClause = ifStatement.elseifClauses();
+    if (elseClause != null) {
+      childStatements = elseClause.statements();
+    } else if (!elseifClause.isEmpty()) {
+      childStatements = elseifClause.get(elseifClause.size() - 1).statements();
+    } else {
+      childStatements = ifStatement.statements();
+    }
+    return getLastStatementOfIf(childStatements.get(childStatements.size() - 1));
   }
 
-  private static StatementTree getLastStatement(IfStatementTree ifStatement) {
-    if (ifStatement.elseClause() == null && ifStatement.elseifClauses().isEmpty()) {
-      return ifStatement.statements().get(0);
+  private void checkStatement(Tree parentStart, Tree parentEnd, StatementTree firstInnerStatement, int nestingStatementNum, List<StatementTree> statements) {
+    if (firstInnerStatement.is(Kind.BLOCK)) {
+      return;
     }
+    int parentLine = ((PHPTree) parentEnd).getLastToken().line();
```

Comment:
you could use methods `line()` and `column()` from the inner class, by extracting them.